pkg/handler/eventhandler.go

```diff
@@ -199,3 +199,23 @@ func (w workqueueWithCustomAddFunc[request]) Add(item request) {
 func isObjectUnchanged[object client.Object](e event.TypedCreateEvent[object]) bool {
 	return e.Object.GetCreationTimestamp().Time.Before(time.Now().Add(-time.Minute))
 }
+
+// addToPriorityQueueCreate adds the reconcile.Request to the priorityqueue in the handler
+// for Create requests if and only if the workqueue being used is of type priorityqueue.PriorityQueue[reconcile.Request]
+func addToPriorityQueueCreate[T client.Object](q priorityqueue.PriorityQueue[reconcile.Request], evt event.TypedCreateEvent[T], item reconcile.Request) {
```

Comment:
Same here, lets just wrap the handler in the constructor

Reply:
These are functions to not repeat the logic in the `TypedEnqueueRequestForObject[T]`  Create/Update as we do export this type.  

Do you want this to have a constructor to wrap with `WithLowPriorityWhenUnchanged` because this will require adding more constraints to deal with the func signature with ^^ to equal `TypedEventHandler[object,request]`. Unless I am misunderstanding something.

Reply:
Sorry I think I just commented on the wrong place here but I see you already updated it 👍